Update Contributing Guide

freearhey opened the pull request:
I decided to completely rewrite the Contributing Guide to make it hopefully more clear. 

If you see any errors or something you think is missing, let me know and I will try to include all edits in the final version.

freearhey committed:
Update CONTRIBUTING.md

freearhey committed:
Update CONTRIBUTING.md

freearhey committed:
Update CONTRIBUTING.md

freearhey committed:
Update CONTRIBUTING.md

freearhey committed:
Update CONTRIBUTING.md

freearhey committed:
Merge branch 'master' into update-contributing-guide

LaneSh4d0w commented:
- Don't repeat yourself when speaking about testing streams and stuff, make a disclaimer like this 

DISCLAIMER : Before doing anything, make sure the link is working in and outside of your country !
- Use a VPN if you got one, and test the link
- Use https://streamtest.in for stream testing

- Not sure if the naming convention was adopted or not, talking about a current subject in this section that could be subject to changes could be misleading.

- A DMCA notice "somethere" ? Simply ask them to create an DMCA Notice issue.
- Repetitive "create an issue". Simply say "Create a stream request issue, create a ... issue"

themak1985 commented on .github/CONTRIBUTING.md:
```diff
@@ -1,52 +1,133 @@
+
+**Would you like us to remove the link to the channel you own the rights to?**
+
+- publish somethere DMCA notice
```
?

themak1985 commented on .github/CONTRIBUTING.md:
```diff
@@ -1,52 +1,133 @@
+- `README.md`: project description generated from the contents of the `.readme/` folder.
+
+## Channel Description Scheme
 
```
Channels are added to playlists using the following format.

themak1985 replied:
Channels should be added to playlists using the following template.

themak1985 commented on .github/CONTRIBUTING.md:
```diff
@@ -1,52 +1,133 @@
+
+**Would you like us to remove the link to the channel you own the rights to?**
+
+- publish somethere DMCA notice
```
Publish your DMCA notice somewhere.

freearhey replied:
Fixed

themak1985 commented on .github/CONTRIBUTING.md:
````diff
@@ -1,52 +1,133 @@
+#EXTINF:-1 tvg-id="EPG_ID" tvg-name="EPG_NAME" tvg-language="PRIMARY_LANGUAGE;SECONDARY_LANGUAGE" tvg-logo="LOGO_URL" group-title="CATEGORY",FULL_NAME STREAM_TIME_SHIFT (ALTERNATIVE_NAME) (STREAM_RESOLUTION) [STREAM_STATUS]
+STREAM_URL
 ```
 
````
## Explanation of Attributes

freearhey replied:
Added.

themak1985 commented on .github/CONTRIBUTING.md:
````diff
@@ -55,119 +136,3 @@ Also, if necessary, you can specify custom HTTP User-Agent or Referrer via `#EXT
 #EXTVLCOPT:http-user-agent=Mozilla/5.0 (Windows NT 10.0; Win64; x64)
 http://example.com/stream.m3u8
 ```
-
````
Why remove this section?

freearhey replied:
Moved the remaining sections to the new version.

freearhey committed:
patch-1

freearhey commented:
@themak1985 Please check the text again. And feel free to edit it yourself if you need to. I have no idea what else to write anyway. 🤷‍♂️

LaneSh4d0w commented on .github/CONTRIBUTING.md:
```diff
@@ -8,12 +8,15 @@ Create an [issue](https://github.com/iptv-org/iptv/issues/new?assignees=&labels=
 - check if the channel is working outside your country. You can use services like [streamtest.in](https://streamtest.in/) to do this.
-- find the file that contains the original link. You can use a [GitHub Search](https://github.com/search/advanced?q=CHANNEL_NAME+repo%3Aiptv-org%2Fiptv+path%3A%2Fchannels&type=Code) to do this (**IMPORTANT:** the file must be in the `channels/` folder).
-- replace the link
+- find out from which country channel is broadcasting. This information can usually be found on [lyngsat.com](https://www.lyngsat.com/search.html) or [wikipedia.org](https://www.wikipedia.org/)
```
from which country "**the**" channel is broadcasting.

LaneSh4d0w commented on .github/CONTRIBUTING.md:
```diff
@@ -38,12 +41,88 @@ In case you were unable to determine which country the channel belongs to, you c
+
+**Do you want to add a new EPG (Electronic Program Guide) source?**
+
+- check which country the EPG is intended for
```
Weird formulation. "Check for which country the EPG is intented for."

Prefer using (maybe) : - check the channels listed in the EPG and check where they're broadcasted.

LaneSh4d0w commented on .github/CONTRIBUTING.md:
```diff
@@ -38,12 +41,88 @@ In case you were unable to determine which country the channel belongs to, you c
+- open the `channels/` folder and find a file with the same name as the country code
+- in the header of the playlist, next to `#EXTM3U`, add an `x-tvg-url` attribute with a link to the EPG source.
+
+The result should look something like this:
```
the result should look like this / the result should be something like this

freearhey committed:
patch-2

LaneSh4d0w commented on .github/CONTRIBUTING.md:
```diff
@@ -12,7 +12,7 @@ Create an [issue](https://github.com/iptv-org/iptv/issues/new?assignees=&labels=
 - make sure that the link you want to add works. It is recommended to use [VLC media player](https://www.videolan.org/vlc/index.html) for this.
 - check if the channel is working outside your country. You can use services like [streamtest.in](https://streamtest.in/) to do this.
-- find out from which country channel is broadcasting. This information can usually be found on [lyngsat.com](https://www.lyngsat.com/search.html) or [wikipedia.org](https://www.wikipedia.org/)
+- find out from which the country channel is broadcasting. This information can usually be found on [lyngsat.com](https://www.lyngsat.com/search.html) or [wikipedia.org](https://www.wikipedia.org/)
```
Wrong position, "from which country **the** channel is being broadcasted."

BellezaEmporium reviewed (commented):
A little mistake, the rest LGTM, waiting for @themak1985 's propositions.

freearhey committed:
patch-3

LaneSh4d0w commented on .github/CONTRIBUTING.md:
```diff
@@ -1,173 +1,221 @@
-Further on the format of links in the playlist. For the broadcast to work in the playlist, you need to add at least minimal information about the channel, like so:
+- make sure that the link you want to add works. It is recommended to use [VLC media player](https://www.videolan.org/vlc/index.html) for this
+- check if the channel is working outside your country. You can use services like [streamtest.in](https://streamtest.in/) to do this
+- find out the full name of the channel and from which country it is broadcasting. This information can usually be found on [lyngsat.com](https://www.lyngsat.com/search.html) or [wikipedia.org](https://www.wikipedia.org/)
```
```suggestion
- find out the full name of the channel and from which country it is being broadcasted. This information can usually be found on [lyngsat.com](https://www.lyngsat.com/search.html) or [wikipedia.org](https://www.wikipedia.org/)
```

LaneSh4d0w commented on .github/CONTRIBUTING.md:
````diff
@@ -1,173 +1,221 @@
-```
-
-But of course, the more channel information you add, the better. Here's an example of what a full version of a link in a playlist might look like:
+In case you were unable to determine which country the channel belongs to, you can put a link to it in the `channels/unsorted.m3u`.
````
```suggestion
In case you were unable to determine which country the channel belongs to, add the channel onto the `channels/unsorted.m3u` playlist.
```

LaneSh4d0w commented on .github/CONTRIBUTING.md:
```diff
@@ -1,173 +1,221 @@
+**Do you want to sort the channels from "channels/unsorted.m3u"?**
 
-More details about each attribute:
+- find out the full name of the channel and from which country it is broadcasting. This information can usually be found on [lyngsat.com](https://www.lyngsat.com/search.html) or [wikipedia.org](https://www.wikipedia.org/)
```
Same as above.
```suggestion
- find out the full name of the channel and from which country it is being broadcasted. This information can usually be found on [lyngsat.com](https://www.lyngsat.com/search.html) or [wikipedia.org](https://www.wikipedia.org/)
```

LaneSh4d0w commented on .github/CONTRIBUTING.md:
```diff
@@ -1,173 +1,221 @@
+- find the file that contains the channel. You can use a [GitHub Search](https://github.com/search/advanced?q=CHANNEL_NAME+repo%3Aiptv-org%2Fiptv+path%3A%2Fchannels&type=Code) to do this.
+- open the file
+- find the channel description
+- specify the appropriate language in the `tvg-language` attribute. The name of the language must comply with [ISO 639-3](https://iso639-3.sil.org/code_tables/639/data?title=&field_iso639_cd_st_mmbrshp_639_1_tid=94671&name_3=&field_iso639_element_scope_tid=All&field_iso639_language_type_tid=51&items_per_page=500) standart.
```
```suggestion
- specify the appropriate language in the `tvg-language` attribute. The name of the language must comply with the [ISO 639-3](https://iso639-3.sil.org/code_tables/639/data?title=&field_iso639_cd_st_mmbrshp_639_1_tid=94671&name_3=&field_iso639_element_scope_tid=All&field_iso639_language_type_tid=51&items_per_page=500) standard.
```

LaneSh4d0w commented on .github/CONTRIBUTING.md:
```diff
@@ -1,173 +1,221 @@
+- send a pull request
 
-To add a new source of EPG (Electronic Program Guide), you must add the `x-tvg-url` attribute to the beginning of the corresponding playlist, like this:
+If a channel is broadcast in several languages at once, you can specify them all through a semicolon, like this:
```
```suggestion
If a channel is broadcasted in several languages at once, you can specify them all through a semicolon, like this:
```

LaneSh4d0w commented on .github/CONTRIBUTING.md:
````diff
@@ -1,173 +1,221 @@
-```
+- check which country the EPG is intended for
+- check that this source is not already listed in the playlist. To do this, find the country in this [table](https://github.com/iptv-org/iptv#playlists-by-country) and see if there is any other link. If not, continue.
+- find the corresponding ISO_3166-2 code for the country. You can find a full list of codes here: https://en.wikipedia.org/wiki/ISO_3166-2
````
Make a direct link to it,
```suggestion
- find the corresponding [ISO_3166-2](https://en.wikipedia.org/wiki/ISO_3166-2) country code.
```

LaneSh4d0w commented on .github/CONTRIBUTING.md:
```diff
@@ -1,173 +1,221 @@
+**Do you want to activate the program guide for the channel?**
+
+- find out which country the channel belongs to. This information can usually be found on [lyngsat.com](https://www.lyngsat.com/search.html) or [wikipedia.org](https://www.wikipedia.org/)
+- find the corresponding ISO_3166-2 code for the country. You can find a full list of codes here: https://en.wikipedia.org/wiki/ISO_3166-2
```
Same as above.
```suggestion
- find the corresponding [ISO_3166-2](https://en.wikipedia.org/wiki/ISO_3166-2) country code.
```

LaneSh4d0w commented on .github/CONTRIBUTING.md:
````diff
@@ -1,173 +1,221 @@
+  ```
+
+- find the channel you are interested in
+- copy it `id` and paste it to the `tvg-id` attribute of the channel description inside the playlist
````
```suggestion
- copy the `id` section of the TVG XML file and paste it into the `tvg-id` attribute of the channel description inside the playlist
```

LaneSh4d0w commented on .github/CONTRIBUTING.md:
```diff
@@ -1,173 +1,221 @@
+
+If you did everything right, then by opening a playlist in a player that supports EPG, you should see the program guide for all updated channels.
+
+In some cases, it may also be necessary to manually specify the source of EPG in the player itself.
```
```suggestion
In some cases, it may also be necessary to manually specify the EPG source in the player itself.
```

LaneSh4d0w commented on .github/CONTRIBUTING.md:
```diff
@@ -1,173 +1,221 @@
+**Did you find a mistake in README.md?**
+
+- open `.readme/template.md`
+- make any necessary changes to the file
```
```suggestion
- make the necessary changes
```

LaneSh4d0w commented on .github/CONTRIBUTING.md:
```diff
@@ -1,173 +1,221 @@
+
+- open `.readme/template.md`
+- make any necessary changes to the file
+- commit all changes
```
It's me being finicky.

```suggestion
- commit the updated README
```

LaneSh4d0w commented on .github/CONTRIBUTING.md:
```diff
@@ -1,173 +1,221 @@
+**Did you find a mistake in this guide?**
+
+- open `.github/CONTRIBUTING.md`
+- make any necessary changes to the file
```
```suggestion
- make the necessary changes
```

LaneSh4d0w commented on .github/CONTRIBUTING.md:
```diff
@@ -1,173 +1,221 @@
+
+- open `.github/CONTRIBUTING.md`
+- make any necessary changes to the file
+- commit all changes
```
```suggestion
- commit the updated guide
```

LaneSh4d0w commented on .github/CONTRIBUTING.md:
````diff
@@ -1,173 +1,221 @@
-```sh
-npm test
+#EXTINF:-1 tvg-id="EPG_ID" tvg-name="EPG_NAME" tvg-language="PRIMARY_LANGUAGE;SECONDARY_LANGUAGE" tvg-logo="LOGO_URL" group-title="CATEGORY",FULL_NAME STREAM_TIME_SHIFT (ALTERNATIVE_NAME) (STREAM_RESOLUTION) [STREAM_STATUS]
+STREAM_URL
````
Move this line and the line above to L185, then explain what the attributes are.

LaneSh4d0w commented on .github/CONTRIBUTING.md:
````diff
@@ -1,173 +1,221 @@
 ```
 
-Another option to exclude specific playlists from tests:
+Also, if necessary, you can specify custom HTTP User-Agent or Referrer via `#EXTVLCOPT` tag:
````
```suggestion
Also, if necessary, you can specify custom HTTP User-Agent or Referrer via the `#EXTVLCOPT` tag:
```

BellezaEmporium reviewed (changes requested):
Finicky changes for some, but those are for clarity purposes. Few little placement mistakes.

freearhey committed:
patch-4

Co-authored-by: LaneSh4d0w <30985701+LaneSh4d0w@users.noreply.github.com>

freearhey committed:
patch-5

Co-authored-by: LaneSh4d0w <30985701+LaneSh4d0w@users.noreply.github.com>

freearhey committed:
patch-6

LaneSh4d0w commented:
LGTM.

BellezaEmporium reviewed (commented):
You forgot the "copy it id" to "copy the id" change.

freearhey committed:
patch-7

themak1985 committed:
Update CONTRIBUTING.md

themak1985 commented:
@freearhey committed some changes, let me know what you think.

freearhey committed:
Removes quotation mark

freearhey committed:
Splits the README.md and CONTRIBUTING.md instructions into two parts

To avoid unnecessary confusion for inexperienced users.

freearhey committed:
Replace "Stream" with "Channel"

That sounds right, doesn't it?

freearhey committed:
Make the header look like the others

freearhey commented:
@themak1985 I guess now we can publish an updated guide.

freearhey committed:
Merge branch 'master' into update-contributing-guide

freearhey committed:
Merge branch 'master' into update-contributing-guide

freearhey committed:
Merge branch 'master' into update-contributing-guide

freearhey committed:
Merge branch 'master' into update-contributing-guide

freearhey committed:
Merge branch 'master' into update-contributing-guide

freearhey committed:
Update Project Structure

freearhey committed:
Update CONTRIBUTING.md

- added table of contents
- added "Add a Category to a channel" section
- updated Channel Description Scheme
- added list of Supported Region Codes
- sort sections

mememe0142 commented:
sorry. where do I request channels?     Family Entertainment Tv             
Alibi 
Antennat tv
CTV sci-fi

thank you

mememe0142 commented:
in ENGLISH please

freearhey commented:
@mememe0142 here: https://github.com/iptv-org/iptv/issues/new?assignees=&labels=channel+request&template=channel-request.md&title=Add%3A+xxx , but before you do, read the [Guide](https://github.com/iptv-org/iptv/blob/master/CONTRIBUTING.md#request-a-channel).
